src/template.cpp

```diff
@@ -479,22 +479,46 @@ bool loadDroidTemplates(const char *filename)
 
 DROID_TEMPLATE *copyTemplate(int player, DROID_TEMPLATE *psTemplate)
 {
-	DROID_TEMPLATE *dup = new DROID_TEMPLATE(*psTemplate);
-	droidTemplates[player][psTemplate->multiPlayerID] = dup;
-	return dup;
+	std::unique_ptr<DROID_TEMPLATE> dup = std::unique_ptr<DROID_TEMPLATE>(new DROID_TEMPLATE(*psTemplate));
+	UDWORD multiPlayerID = dup->multiPlayerID;
+	auto result = droidTemplates[player].insert(std::pair<UDWORD, std::unique_ptr<DROID_TEMPLATE>>(multiPlayerID, std::move(dup)));
+	return result.first->second.get();
 }
 
-void addTemplate(int player, DROID_TEMPLATE *psTemplate)
+DROID_TEMPLATE* addTemplate(int player, std::unique_ptr<DROID_TEMPLATE> psTemplate)
 {
-	droidTemplates[player][psTemplate->multiPlayerID] = psTemplate;
+	UDWORD multiPlayerID = psTemplate->multiPlayerID;
+	auto result = droidTemplates[player].insert(std::pair<UDWORD, std::unique_ptr<DROID_TEMPLATE>>(multiPlayerID, std::move(psTemplate)));
+	return result.first->second.get();
 }
 
-void clearTemplates(int player)
+void enumerateTemplates(int player, const std::function<bool (DROID_TEMPLATE* psTemplate)>& func)
 {
 	for (auto &keyvaluepair : droidTemplates[player])
 	{
-		delete keyvaluepair.second;
+		if (!func(keyvaluepair.second.get()))
+		{
+			break;
+		}
+	}
+}
+
+DROID_TEMPLATE* findPlayerTemplateById(int player, UDWORD templateId)
+{
+	if (droidTemplates[player].count(templateId) > 0)
```

Comment:
Is there an advantage in using `count` instead of `find`?

Reply:
Good catch - fixed. 👍